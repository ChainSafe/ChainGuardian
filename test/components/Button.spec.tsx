import * as React from "react";
import * as renderer from "react-test-renderer";
import {
    ButtonPrimitive, 
    ButtonPrimary, 
    ButtonSecondary, 
    ButtonInverted, 
    ButtonDestructive} from "../../src/renderer/components/Button/ButtonStandard";
import {
    BackTab, 
    CopyButton, 
    AddButton, 
    BackButton} from "../../src/renderer/components/Button/ButtonAction";

describe("ButtonPrimitive", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<ButtonPrimitive>Submit</ButtonPrimitive>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders disabled button", () => {
        const tree = renderer
            .create(<ButtonPrimitive disabled={true}>Submit</ButtonPrimitive>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders focused button", () => {
        const tree = renderer
            .create(<ButtonPrimitive focused={true}>Submit</ButtonPrimitive>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("ButtonPrimary", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<ButtonPrimary>Submit</ButtonPrimary>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders disabled button", () => {
        const tree = renderer
            .create(<ButtonPrimary disabled={true}>Submit</ButtonPrimary>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders focused button", () => {
        const tree = renderer
            .create(<ButtonPrimary focused={true}>Submit</ButtonPrimary>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("ButtonSecondary", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<ButtonSecondary>Submit</ButtonSecondary>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders disabled button", () => {
        const tree = renderer
            .create(<ButtonSecondary disabled={true}>Submit</ButtonSecondary>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders focused button", () => {
        const tree = renderer
            .create(<ButtonSecondary focused={true}>Submit</ButtonSecondary>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("ButtonInverted", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<ButtonInverted>Submit</ButtonInverted>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders disabled button", () => {
        const tree = renderer
            .create(<ButtonInverted disabled={true}>Submit</ButtonInverted>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders focused button", () => {
        const tree = renderer
            .create(<ButtonInverted focused={true}>Submit</ButtonInverted>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("ButtonDestructive", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<ButtonDestructive>Submit</ButtonDestructive>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders disabled button", () => {
        const tree = renderer
            .create(<ButtonDestructive disabled={true}>Submit</ButtonDestructive>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders focused button", () => {
        const tree = renderer
            .create(<ButtonDestructive focused={true}>Submit</ButtonDestructive>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("BackTab", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<BackTab />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("CopyButton", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<CopyButton />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("AddButton", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<AddButton />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("BackButton", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<BackButton />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});