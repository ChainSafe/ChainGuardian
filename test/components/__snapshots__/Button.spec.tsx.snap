// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton renders correctly 1`] = `
<button
  className="add-button"
>
  <img
    className="icon"
    src="test-file-stub"
  />
</button>
`;

exports[`BackButton renders correctly 1`] = `
<button
  className="back-button"
>
  <img
    className="icon"
    src="test-file-stub"
  />
</button>
`;

exports[`BackTab renders correctly 1`] = `
<button
  className="back-tab"
>
  <img
    className="icon"
    src="test-file-stub"
  />
</button>
`;

exports[`ButtonDestructive renders correctly 1`] = `
<button
  className="button btn-destructive"
>
  Submit
</button>
`;

exports[`ButtonDestructive renders disabled button 1`] = `
<button
  className="button btn-destructive"
  disabled={true}
>
  Submit
</button>
`;

exports[`ButtonDestructive renders focused button 1`] = `
<button
  className="button btn-destructive focused"
>
  Submit
</button>
`;

exports[`ButtonInverted renders correctly 1`] = `
<button
  className="button btn-inverted"
>
  Submit
</button>
`;

exports[`ButtonInverted renders disabled button 1`] = `
<button
  className="button btn-inverted"
  disabled={true}
>
  Submit
</button>
`;

exports[`ButtonInverted renders focused button 1`] = `
<button
  className="button btn-inverted focused"
>
  Submit
</button>
`;

exports[`ButtonPrimary renders correctly 1`] = `
<button
  className="button btn-primary"
>
  Submit
</button>
`;

exports[`ButtonPrimary renders disabled button 1`] = `
<button
  className="button btn-primary"
  disabled={true}
>
  Submit
</button>
`;

exports[`ButtonPrimary renders focused button 1`] = `
<button
  className="button btn-primary focused"
>
  Submit
</button>
`;

exports[`ButtonPrimitive renders correctly 1`] = `
<button
  className="button btn-primitive"
>
  Submit
</button>
`;

exports[`ButtonPrimitive renders disabled button 1`] = `
<button
  className="button btn-primitive"
  disabled={true}
>
  Submit
</button>
`;

exports[`ButtonPrimitive renders focused button 1`] = `
<button
  className="button btn-primitive focused"
>
  Submit
</button>
`;

exports[`ButtonSecondary renders correctly 1`] = `
<button
  className="button btn-secondary"
>
  Submit
</button>
`;

exports[`ButtonSecondary renders disabled button 1`] = `
<button
  className="button btn-secondary"
  disabled={true}
>
  Submit
</button>
`;

exports[`ButtonSecondary renders focused button 1`] = `
<button
  className="button btn-secondary focused"
>
  Submit
</button>
`;

exports[`CopyButton renders correctly 1`] = `
<button
  className="copy-button"
>
  <img
    className="icon copy"
    src="test-file-stub"
  />
  <img
    className="icon copyall"
    src="test-file-stub"
  />
  <img
    className="icon copied"
    src="test-file-stub"
  />
</button>
`;
